### PR TITLE
`Iris`: New message indicator

### PR DESCRIPTION
**Pull request by dakennguyen**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [ ] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [ ] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [ ] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [ ] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [ ] I documented the Java code using JavaDoc style.
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

Show a red dot to indicate new messages. When user clicks on the chatbot icon, reposition user to the point that has new messages

### Description

- Whenever receiving a response from the server, increase `numNewMessages` by 1
- When close the chatbot dialog, reset `numNewMessages` to 0
- If chatbot is closed and `numNewMessages` > 0 then show the red dot
- If chatbot is opened, and `numNewMessage` > 0 then scroll to the unread message point. Otherwise, scroll to the bottom

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise with Iris enabled

1. Log in to Artemis
2. Navigate to a Programming Exercise details page
3. Open chatbot
4. Throttle network and chat something, expect to see a "typing" indicator
5. Close the chatbot and wait until the response comes. Expect to see a red dot to indicate that there are unread messages
6. Open the chatbot and expect to be scrolled to the new message with a "unread message" line
7. Chat something and the "unread messages" line should stay at the same location
8. Close the chatbot and open it again and should be scrolled to the bottom, and "unread messages" line should be disappeared

Note: close the chatbot by either pressing the escape key, clicking on the "x" icon, or clicking on the chatbot icon. They should behave the same

#### Performance Review
- [ ] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Test Coverage

| File                        | Lines           |Confirmation (assert/expect) |
|-----------------------------|-----------------|---|
| state-store.service.ts      | 95.34%		 |✅|
| exercise-chatbot.component.ts     | 90.56%		 |✅|
| exercise-chat-widget.component.ts        | 92.45%		 |✅|

### Screenshots

https://github.com/ls1intum/Artemis/assets/18320285/713c02ce-1a3e-4cd8-b789-256b877aed6a

### CODE REVIEW

**Review by eceeeren — DISMISSED** [review]
Tested on TS9, works fine!

**Review by asliayk — APPROVED** [review]
tested on ts9, worked as expected

File: src/test/javascript/spec/component/iris/ui/exercise-chatbot.component.spec.ts [hunk 1]
```diff
@@ -38,7 +40,9 @@ describe('ExerciseChatbotComponent', () => {
 
         mockDialog = {
             open: jest.fn().mockReturnValue({
-                afterClosed: jest.fn().mockReturnValue(of('true')),
```
Comment (JohannesStoehr) [inline src/test/javascript/spec/component/iris/ui/exercise-chatbot.component.spec.ts:41]: Unused import because of this

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Code looks good, but you missed one import:

**Review by DemjenDavid — DISMISSED** [review]
Tested on ts9 on 2 browsers

**Review by RY997 — DISMISSED** [review]
Worked as expected. For me it would be better to adjust the upper and lower padding/margin when displaying "UNREAD MESSAGE".
image

**Review by greekgeek1 — DISMISSED** [review]
Tested on ts9. Works as expencted

**Review by SabrinaGlatz — APPROVED** [review]
tested on ts9. It works as expected, but I have I have one small  comment: 
I left the page while iris answered my question. When I returned, I didn't see the indicator anymore. Could be added, but I'm not sure if it's necessary.

File: src/test/javascript/spec/component/iris/ui/exercise-chatbot.component.spec.ts (hunk 1 above)
Comment (dakennguyen) [reply src/test/javascript/spec/component/iris/ui/exercise-chatbot.component.spec.ts:41]: thank you! that was a good catch!

**Review by asliayk — APPROVED** [review]
reapproved

**Review by MichaelOwenDyer — APPROVED** [review]

**Review by undernagruzez — APPROVED** [review]
LGTM

**Review by RY997 — APPROVED** [review]
Looks good!

**Review by JohannesStoehr — APPROVED** [review]
New code looks good to me

**Pull request merged**